cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2771 @@
+  std::map<std::string_view, Value>::const_iterator find(const std::string_view key) const {
+    return property_map_.find(key);
+  }
+
+  /// @exception std::runtime_error Map contains value(s) of unknown type.
+  bool operator==(const Properties &other) const;
+  /// @exception std::runtime_error Map contains value(s) of unknown type.
+  bool operator!=(const Properties &other) const { return !(*this == other); }
+
+ private:
+  std::map<const std::string_view, Value> property_map_;
+};
+
+/// @brief View of node labels.
+class Labels {
+ public:
+  explicit Labels(mgp_vertex *node_ptr) : node_ptr_(node_ptr) {}
+
+  /// @brief Returns the number of the labels, i.e. the size of their list.
+  size_t Size() const { return mgp::vertex_labels_count(node_ptr_); }
+
+  /// @brief Return the node’s label at position `index`.
+  std::string_view operator[](size_t index) const;
+
+  class Iterator {
+   private:
+    friend class Labels;
+
+   public:
+    bool operator==(const Iterator &other) const { return iterable_ == other.iterable_ && index_ == other.index_; }
+
+    bool operator!=(const Iterator &other) const { return !(*this == other); }
+
+    Iterator &operator++() {
+      index_++;
+      return *this;
+    }
+
+    const std::string_view operator*() const;
+
+   private:
+    Iterator(const Labels *iterable, size_t index) : iterable_(iterable), index_(index) {}
+
+    const Labels *iterable_;
+    size_t index_;
+  };
+
+  Iterator begin() { return Iterator(this, 0); }
+  Iterator end() { return Iterator(this, Size()); }
+
+  Iterator cbegin() { return Iterator(this, 0); }
+  Iterator cend() { return Iterator(this, Size()); }
+
+ private:
+  mgp_vertex *node_ptr_;
+};
+/* #endregion */
+
+/* #region Types */
+
+/* #region Containers (List, Map) */
+
+/// @brief Wrapper class for @ref mgp_list.
+class List {
+ private:
+  friend class Value;
+  friend class Record;
+  friend class Result;
+  friend struct Parameter;
+
+ public:
+  /// @brief Creates a List from the copy of the given @ref mgp_list.
+  explicit List(mgp_list *ptr) : ptr_(mgp::list_copy(ptr, memory)) {}
+  /// @brief Creates a List from the copy of the given @ref mgp_list.
+  explicit List(const mgp_list *const_ptr) : ptr_(mgp::list_copy(const_cast<mgp_list *>(const_ptr), memory)) {}
+
+  /// @brief Creates an empty List.
+  explicit List() : ptr_(mgp::list_make_empty(0, memory)) {}
+
+  /// @brief Creates a List with the given `capacity`.
+  explicit List(size_t capacity) : ptr_(mgp::list_make_empty(capacity, memory)) {}
+
+  /// @brief Creates a List from the given vector.
+  explicit List(const std::vector<Value> &values);
+  /// @brief Creates a List from the given vector.
+  explicit List(std::vector<Value> &&values);
+
+  /// @brief Creates a List from the given initializer_list.
+  explicit List(const std::initializer_list<Value> list);
+
+  List(const List &other) : List(other.ptr_) {}
```

Comment:
If you copy owning pointers, then you will end up in double free situations. I haven't spotted this on the mage repo, only as I checked a memory leak that @Josipmrden just fixed. This is true for all copy constructors. In my opinion best solution is to call `mgp_xxx_copy` to copy the underlying object.